### ISSUE TITLE
can I train a model using irregular mask with this project?

### ISSUE THREAD

**Issue by muyue1238** [JiahuiYu/generative_inpainting#197]
Hi there, I'd like to know if I'll achieve good performance using irregular masks during training process? I've tested the model you provided and its performance on irregular masks is a little worse than on regular masks. So I want to retrain a model with irregular masks. The fact is that I'm not sure if its ok to use irregular ones as input, can you answer this question?

FYI, looking forward to the release of Deep Fill V2.

**Comment by JiahuiYu** [comment]
Hi, I encourage you to have a try.

**Comment by muyue1238** [comment]
Hi again， thanks  for your quick response. I've encountered a problem and wonder if you please help me. I'm trying to use this kind of masks while training, 
![test](https://user-images.githubusercontent.com/16850896/50196546-7e39f180-037e-11e9-812e-7742be63cd5c.jpg)
and I've changed some code you provided, somehow it didn't work. Errors like below always came up:
`Caused by op 'inpaint_net_1/resize/ResizeNearestNeighbor', defined at:
  File "train.py", line 88, in <module>
    'model': model, 'data': data, 'config': config, 'loss_type': 'd'},
  File "/home/xwbian/anaconda2/envs/tf-py3/lib/python3.6/site-packages/neuralgym/callbacks/secondary_trainer.py", line 22, in __init__
    Trainer.__init__(self, primary=False, **context)
  File "/home/xwbian/anaconda2/envs/tf-py3/lib/python3.6/site-packages/neuralgym/train/trainer.py", line 41, in __init__
    self._train_op, self._loss = self.train_ops_and_losses()
  File "/home/xwbian/anaconda2/envs/tf-py3/lib/python3.6/site-packages/neuralgym/train/trainer.py", line 250, in train_ops_and_losses
    loss = self.context['graph_def'](**graph_def_kwargs)
  File "train.py", line 23, in multigpu_graph_def
    images, config, reuse=True)
  File "/YeDisk/home/xwbian/projects/image_impaiting/generative_inpainting/inpaint_model.py", line 155, in build_graph_with_losses
    padding=config.PADDING)
  File "/YeDisk/home/xwbian/projects/image_impaiting/generative_inpainting/inpaint_model.py", line 56, in build_inpaint_net
    mask_s = resize_mask_like(mask, x)
  File "/YeDisk/home/xwbian/projects/image_impaiting/generative_inpainting/inpaint_ops.py", line 255, in resize_mask_like
    func=tf.image.resize_nearest_neighbor)
  File "/home/xwbian/anaconda2/envs/tf-py3/lib/python3.6/site-packages/neuralgym/ops/layers.py", line 128, in resize
    align_corners=align_corners)
  File "/home/xwbian/anaconda2/envs/tf-py3/lib/python3.6/site-packages/tensorflow/python/ops/gen_image_ops.py", line 2506, in resize_nearest_neighbor
    align_corners=align_corners, name=name)
  File "/home/xwbian/anaconda2/envs/tf-py3/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py", line 787, in _apply_op_helper
    op_def=op_def)
  File "/home/xwbian/anaconda2/envs/tf-py3/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 3290, in create_op
    op_def=op_def)
  File "/home/xwbian/anaconda2/envs/tf-py3/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 1654, in __init__
    self._traceback = self._graph._extract_stack()  # pylint: disable=protected-access

InvalidArgumentError (see above for traceback): input must be 4-dimensional[256,256,1]
	 [[Node: inpaint_net_1/resize/ResizeNearestNeighbor = ResizeNearestNeighbor[T=DT_FLOAT, align_corners=true, _device="/job:localhost/replica:0/task:0/device:CPU:0"](mask_c_1/PyFuncStateless, inpaint_net_1/resize/ResizeNearestNeighbor/size)]]`
I know the information I provided is not very much, could you tell which part approximately I'm wrong? Thanks.

**Comment by muyue1238** [comment]
never mind, kinda solved it myself. If it turns out OK, I'll show it here.
